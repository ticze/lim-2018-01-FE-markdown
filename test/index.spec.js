const mdLinks = require('../index');
// jest.setTimeout(10000); 
const options = {
  validate: false,
  stats: false
}

test('mdLinks deberia ser Promise', () => {
	return expect(mdLinks('test/src/index.md', options)).toBeInstanceOf(Promise);
});

test('resolver mdLinks(test/src/index.md, options) -> options.stats = true && options.validate = false', () => {
	const options ={
		validate : false,
		stats : true
	}
 return expect(Promise.resolve(mdLinks('test/src/index.md', options))).resolves.toMatchSnapshot();
});

test('resolver mdLinks(test/src/index.md, options) -> options.stats = false && options.validate = true', () => {
	const options ={
		validate : true,
		stats : false
	}
 return expect(Promise.resolve(mdLinks('test/src/index.md', options))).resolves.toEqual([ { link: 'https://github.com/AnaSalazar/curricula-js/blob/04-social-network/04-social-network/02-jquery/08-code-challenges/foodmap/2.jpg?raw=true',
 status: 200 },
{ link: 'https://AnaSalazar/curricula-js/blob/04-social-network/04-social-network/02-jquery/08-code-challenges/foodmap/splash.jpg?raw=true',
 status: 0 } ]);
});

test('resolver mdLinks(test/src/index.md, options) -> options.stats = true && options.validate = true', () => {
	const options ={
		validate : true,
		stats : true
	}
 return expect(Promise.resolve(mdLinks('test/src/index.md', options))).resolves.toMatchSnapshot();
});