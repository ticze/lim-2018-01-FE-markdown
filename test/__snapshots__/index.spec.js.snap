// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolver mdLinks(test/src/index.md, options) -> options.stats = true && options.validate = false 1`] = `
Array [
  undefined,
]
`;

exports[`resolver mdLinks(test/src/index.md, options) -> options.stats = true && options.validate = true 1`] = `
Array [
  undefined,
]
`;
